update packages for npm warnings

huss opened the pull request:
# Description

This updates packages to address some of the warnings during an OED install from the npm install. It helped with:
- Update chai-http that allowed update of superagent. This gets OED to v8 which is not too old verses the most recent v9. npm still warns about v8 being deprecated. This also fixes the warning about formidable that came from superagent by getting to v2.
- Update multer to latest lts. This fixes the warnings.

A number of other packages were modified as part of this process. Testing has not found any issue.

## Type of change

- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

## Limitations

It does not fix these:
- "mumath@3.3.4: Redundant dependency in your project". This is used by plotly.js & react-plotly and going to the latest versions did not help so not done. This issue (https://github.com/colorjs/color-space/issues/50) may relate to the problem and it is hoped once other packages upgrade usage this will go away.
- core-js relates to rc-slider and babel. A commit to PR #1249 should resolve the warning and remove this package from OED.

This does not do a general package update.

huss committed:
update multer package

huss committed:
upgrade chai-http

This gets OED to a fairly recent version of superagent but still warned
to upgrade. It also resolves the warning for formidable.

ChrisMart21 commented:
I'm seeing the following warning.

>  WARN deprecated superagent@8.1.2: Please upgrade to v9.0.0+ as we have fixed ....

This PR has sat for a while, and I'm wondering if this is something that was missed or may be new that should be looked into.

huss commented:
> I'm seeing the following warning.
> 
> > WARN deprecated superagent@8.1.2: Please upgrade to v9.0.0+ as we have fixed ....
> 
> This PR has sat for a while, and I'm wondering if this is something that was missed or may be new that should be looked into.

Thanks for asking. I went back to my notes (glad I kept them while I worked on this :-) and found this:

chai-http@4.3.0 is 5 years old. chai-http@4.4.0 is latest stable from 1 year ago so try installing that. This got us to superagent@8.1.2 but still warning for deprecated for v9.0.0+ but will have to live with this. May also want to upgrade chai. It also update the @types and changed other packages. It passed all tests so seems okay.

So, when I did this there was not a version of the package that had this as a dependency that could get us to superagent@9. For now, I think we can let it go and hope that the next package sweep/update will enable this.

huss committed:
Merge remote-tracking branch 'upstream/development' into packages2406

huss committed:
fix new unneeded package @types/history

huss commented:
I saw this now had a merge conflict so I fixed it up. I hope I did not leave any unneeded packages.

I noticed that this created a new unneeded package on @types/history so I fixed that too. This is a new change.

huss commented:
> > I'm seeing the following warning.
> > > WARN deprecated superagent@8.1.2: Please upgrade to v9.0.0+ as we have fixed ....
> > 
> > 
> > This PR has sat for a while, and I'm wondering if this is something that was missed or may be new that should be looked into.
> 
> Thanks for asking. I went back to my notes (glad I kept them while I worked on this :-) and found this:
> 
> chai-http@4.3.0 is 5 years old. chai-http@4.4.0 is latest stable from 1 year ago so try installing that. This got us to superagent@8.1.2 but still warning for deprecated for v9.0.0+ but will have to live with this. May also want to upgrade chai. It also update the @types and changed other packages. It passed all tests so seems okay.
> 
> So, when I did this there was not a version of the package that had this as a dependency that could get us to superagent@9. For now, I think we can let it go and hope that the next package sweep/update will enable this.

The recent PR #1294 does upgrade superagent but is having issues that need to resolve.
